packages/astro/src/i18n/index.ts

```diff
@@ -92,6 +113,9 @@ export function getLocaleAbsoluteUrlList({ site, ...rest }: GetLocaleAbsoluteUrl
  * - replaces the `_` with a `-`;
  * - transforms all letters to be lower case;
  */
-function normalizeLocale(locale: string): string {
+function normalizeTheLocale(locale: string, shouldNormalize: boolean): string {
+	if (!shouldNormalize) {
+		return locale;
+	}
 	return locale.replaceAll('_', '-').toLowerCase();
 }
```

Comment:
nit: I find passing `shouldNormalize` to determine whether the function is a no-op a bit strange of a pattern. And instead the callee should probably decide if they should call this function in the first place?

Not a big deal for me that needs changing but thought I point it out.

Reply:
Yeah, I was on the fence on this one. I didn't want to add too much branching in the callee, but I can address this comment in the next PR. Thank you